### PR TITLE
Add shared memory heap to memory scan

### PR DESCRIPTION
**Pull request by ktrzcinx**
Newly added heap should be taken into account during memory usage
scanning.

### CODE REVIEW

File: src/ipc/handler.c [hunk 1]
```diff
@@ -1302,11 +1302,18 @@ static int fill_mem_usage_elems(enum mem_zone zone, enum sof_ipc_dbg_mem_zone ip
 	return elem_number;
 }
 
+#if CONFIG_CORE_COUNT > 1
+#define PLATFORM_HEAP_SYSTEM_SHARED_CNT (PLATFORM_HEAP_SYSTEM_SHARED + PLATFORM_HEAP_RUNTIME_SHARED)
```
Comment (xiulipan) [inline src/ipc/handler.c:1306]: maybe put this define into the same file with `PLATFORM_HEAP_SYSTEM_SHARED `

**Review by xiulipan — COMMENTED** [review]
Any driver side change needed? I think at least an ABI alignment is needed?

File: src/ipc/handler.c (hunk 1 above)
Comment (ktrzcinx) [reply src/ipc/handler.c:1306]: heap counters are platform specific values, while this one is more generic (common for each multi-core platform), so keeping then in the same place doesn't sound like a good idea.

**Comment by lgirdwood** [comment]
@ktrzcinx would this be a debug ABI update only ?

**Comment by ktrzcinx** [comment]
> @ktrzcinx would this be a debug ABI update only ?

I'm not sure - it's debug feature but those fields are defined also in kernel code.

**Comment by lgirdwood** [comment]
@kv2019i any comment - does kernel see this as ABI or debug ABI update ?

**Comment by kv2019i** [comment]
@lgirdwood wrote:
> @kv2019i any comment - does kernel see this as ABI or debug ABI update ?

Kernel can handle new zones within existing ABI, so we can proceed with this.

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
@kv2019i good to go now for kernel side.

**Pull request merged**